keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionCheckerTests.kt

```diff
@@ -333,5 +351,102 @@ internal class NewEnvironmentPromotionCheckerTests : JUnit5Minutests {
         }
       }
     }
+
+    context("config contains multiple types of artifacts") {
+      before {
+        every {
+          repository.artifactVersions(dockerArtifact)
+        } returns listOf("2.0")
+
+        every {
+          repository.artifactVersions(debianArtifact)
+        } returns listOf("3.0")
+
+        every {
+          repository.getQueuedConstraintApprovals(deliveryConfigWith2ArtifactTypes.name, environment.name, dockerArtifact.type)
+        } returns setOf("2.0")
+
+        every {
+          repository.getQueuedConstraintApprovals(deliveryConfigWith2ArtifactTypes.name, environment.name, debianArtifact.type)
+        } returns setOf("3.0")
+
+        every {
+          environmentConstraintRunner.checkStatelessConstraints(dockerArtifact, deliveryConfigWith2ArtifactTypes, "2.0", environment)
+        } returns true
+
+        every {
+          environmentConstraintRunner.checkStatelessConstraints(debianArtifact, deliveryConfigWith2ArtifactTypes, "3.0", environment)
+        } returns true
+
+        every {
+          repository.approveVersionFor(deliveryConfigWith2ArtifactTypes, dockerArtifact, "2.0", environment.name)
+        } returns true
+
+        every {
+          repository.approveVersionFor(deliveryConfigWith2ArtifactTypes, debianArtifact, "3.0", environment.name)
+        } returns true
+
+        every {
+          repository.pinnedEnvironments(any())
+        } returns emptyList()
+
+        every {
+          repository.vetoedEnvironmentVersions(any())
+        } returns emptyList()
+
+        runBlocking {
+          subject.checkEnvironments(deliveryConfigWith2ArtifactTypes)
+        }
+      }
+
+      test("verify the right artifact type and version approved") {
+        verify {
+          repository.approveVersionFor(deliveryConfigWith2ArtifactTypes, dockerArtifact, "2.0", environment.name)
+          repository.approveVersionFor(deliveryConfigWith2ArtifactTypes, debianArtifact, "3.0", environment.name)
+        }
+      }
+
+      test("approved versions of other artifact types are not getting mixed up and are not checked") {
```

Comment:
Prior to this change, this test failed as different artifact types and versions got checked.